--- origsrc/libdvdread-4.1.3/src/dvd_input.c	2008-09-06 16:55:51.000000000 -0500
+++ src/libdvdread-4.1.3/src/dvd_input.c	2009-03-30 00:00:05.453125000 -0500
@@ -164,7 +164,7 @@ static dvd_input_t file_open(const char 
   }
 
   /* Open the device */
-#ifndef WIN32
+#ifndef O_BINARY
   dev->fd = open(target, O_RDONLY);
 #else
   dev->fd = open(target, O_RDONLY | O_BINARY);
@@ -284,6 +284,8 @@ int dvdinput_setup(void)
 
 #ifdef __APPLE__
   #define CSS_LIB "libdvdcss.2.dylib"
+#elif defined(__CYGWIN__)
+  #define CSS_LIB "cygdvdcss-2.dll"
 #elif defined(WIN32)
   #define CSS_LIB "libdvdcss.dll"
 #else
