--- origsrc/libdvdread-5.0.3/Makefile.am	2014-08-28 01:41:51.000000000 -0500
+++ src/libdvdread-5.0.3/Makefile.am	2015-05-19 15:08:18.399151800 -0500
@@ -34,6 +34,7 @@ libdvdread_la_SOURCES = \
 libdvdread_la_LIBADD = $(CSS_LIBS)
 
 libdvdread_la_LDFLAGS = -version-info $(DVDREAD_LTVERSION) \
+	-no-undefined \
 	-export-symbols-regex "(^dvd.*|^nav.*|^ifo.*|^DVD.*|^UDF.*)"
 
 pkgincludedir = $(includedir)/dvdread
--- origsrc/libdvdread-5.0.3/src/bswap.h	2014-10-01 07:54:12.000000000 -0500
+++ src/libdvdread-5.0.3/src/bswap.h	2015-05-19 15:05:37.447213500 -0500
@@ -37,7 +37,7 @@
 #include <sys/param.h>
 #endif
 
-#if defined(__linux__) || defined(__GLIBC__)
+#if defined(__linux__) || defined(__GLIBC__) || defined(__CYGWIN__)
 #include <byteswap.h>
 #define B2N_16(x) x = bswap_16(x)
 #define B2N_32(x) x = bswap_32(x)
--- origsrc/libdvdread-5.0.3/src/dvd_input.c	2015-01-28 10:17:27.000000000 -0600
+++ src/libdvdread-5.0.3/src/dvd_input.c	2015-05-19 15:05:37.452214100 -0500
@@ -185,7 +185,7 @@ static dvd_input_t file_open(const char
   }
 
   /* Open the device */
-#if !defined(WIN32) && !defined(__OS2__)
+#ifndef O_BINARY
   dev->fd = open(target, O_RDONLY);
 #else
   dev->fd = open(target, O_RDONLY | O_BINARY);
@@ -303,6 +303,8 @@ int dvdinput_setup(void)
 
 #ifdef __APPLE__
   #define CSS_LIB "libdvdcss.2.dylib"
+#elif defined(__CYGWIN__)
+  #define CSS_LIB "cygdvdcss-2.dll"
 #elif defined(WIN32)
   #define CSS_LIB "libdvdcss-2.dll"
 #elif defined(__OS2__)
--- origsrc/libdvdread-5.0.3/src/dvd_reader.c	2015-03-14 13:09:18.000000000 -0500
+++ src/libdvdread-5.0.3/src/dvd_reader.c	2015-05-19 15:05:37.459715100 -0500
@@ -65,7 +65,7 @@ static inline int _private_gettimeofday(
 # include <sys/mount.h>
 #elif defined(SYS_BSD)
 # include <fstab.h>
-#elif defined(__linux__)
+#elif defined(__linux__) || defined(__CYGWIN__)
 # include <mntent.h>
 # include <paths.h>
 #endif
@@ -435,7 +435,7 @@ static dvd_reader_t *DVDOpenCommon( cons
     dvd_reader_t *auth_drive = 0;
 #if defined(SYS_BSD)
     struct fstab* fe;
-#elif defined(__sun) || defined(__linux__)
+#elif defined(__sun) || defined(__linux__) || defined(__CYGWIN__)
     FILE *mntfile;
 #endif
 
@@ -553,7 +553,7 @@ static dvd_reader_t *DVDOpenCommon( cons
       }
       fclose( mntfile );
     }
-#elif defined(__linux__)
+#elif defined(__linux__) || defined(__CYGWIN__)
     mntfile = fopen( _PATH_MOUNTED, "r" );
     if( mntfile ) {
       struct mntent *me;
