--- origsrc/libdvdread-0.9.7/configure.in	2006-10-06 03:12:31.000000000 -0500
+++ src/libdvdread-0.9.7/configure.in	2006-10-09 14:19:35.547250000 -0500
@@ -103,15 +103,15 @@
       AC_MSG_ERROR(You need libdvdcss (dvdcss.h))
     )
     AC_MSG_CHECKING([for dvdcss_interface_2 in -ldvdcss])
-    saved_LDFLAGS=$LDFLAGS
-    LDFLAGS="$LDFLAGS -ldvdcss"
+    saved_LIBS=$LIBS
+    LIBS="$LIBS -ldvdcss"
     AC_TRY_LINK([#include <dvdcss/dvdcss.h>], 
                 [if(!dvdcss_interface_2) { return 1; }],
         [ CSS_LIBS=-ldvdcss ],
         [ AC_MSG_RESULT([no])
           AC_MSG_ERROR(You need libdvdcss.so.2 or later)
         ] )
-    LDFLAGS=$saved_LDFLAGS
+    LIBS=$saved_LIBS
     AC_MSG_RESULT([yes])
   else
     dnl -w added to shutup GCC3.1's cpp warning about -I/usr/local
--- origsrc/libdvdread-0.9.7/dvdread/Makefile.am	2006-10-06 03:12:31.000000000 -0500
+++ src/libdvdread-0.9.7/dvdread/Makefile.am	2006-10-09 14:19:14.328500000 -0500
@@ -6,7 +6,7 @@
 	dvdread_internal.h cmd_print.c \
 	md5.c md5.h
 libdvdread_la_LIBADD = @DL_LIBS@ @CSS_LIBS@
-libdvdread_la_LDFLAGS = -version-info 5:1:2 -export-symbols-regex "(DVD.*|ifo.*|nav.*|cmd.*)"
+libdvdread_la_LDFLAGS = -version-info 5:1:2 -no-undefined -export-symbols-regex "(DVD.*|ifo.*|nav.*|cmd.*)"
 
 pkgincludedir = $(includedir)/dvdread
 
--- origsrc/libdvdread-0.9.7/dvdread/bswap.h	2006-06-06 15:03:37.000000000 -0500
+++ src/libdvdread-0.9.7/dvdread/bswap.h	2006-10-09 14:19:35.578500000 -0500
@@ -34,7 +34,7 @@
 #include <sys/param.h>
 #endif
 
-#if defined(__linux__)
+#if defined(__linux__) || defined(__CYGWIN__)
 #include <byteswap.h>
 #define B2N_16(x) x = bswap_16(x)
 #define B2N_32(x) x = bswap_32(x)
--- origsrc/libdvdread-0.9.7/dvdread/dvd_reader.c	2006-10-06 02:58:03.000000000 -0500
+++ src/libdvdread-0.9.7/dvdread/dvd_reader.c	2006-10-09 14:19:35.609750000 -0500
@@ -41,7 +41,7 @@
 #include <sys/mnttab.h>
 #elif defined(SYS_BSD)
 #include <fstab.h>
-#elif defined(__linux__)
+#elif defined(__linux__) || defined(__CYGWIN__)
 #include <mntent.h>
 #endif
 
@@ -494,7 +494,7 @@
     char *path_copy;
 #if defined(SYS_BSD)
     struct fstab* fe;
-#elif defined(__sun) || defined(__linux__)
+#elif defined(__sun) || defined(__linux__) || defined(__CYGWIN__)
     FILE *mntfile;
 #endif
 
@@ -598,7 +598,7 @@
       }
       fclose( mntfile );
     }
-#elif defined(__linux__)
+#elif defined(__linux__) || defined(__CYGWIN__)
     mntfile = fopen( MOUNTED, "r" );
     if( mntfile ) {
       struct mntent *me;
